### PR TITLE
[issue#6461] Fix cargo commands list

### LINKED ISSUE #6461: Bash autocomplete does not work correctly
**Problem**
In bash, when you pressing <TAB> twice you'll get non-exist commands in autocomplete (such a, and, all, etc)

**Steps**

1. run bash
2. write `cargo ` and press <TAB> key twice
3. check output:

```
$ cargo
Display all 148 possibilities? (y or n)
(src/main.rs)      binary             crate              file               If                 local              not                qualified          Save               uninstall
<path>             build              crates.io          files              in                 locally.           of                 read               search             unit
a                  Build              Create             fix                including          locate-project     on                 read-manifest      Search             update
all                by                 custom             flags.             index              location           Output             recorded           Show               Update
an                 cargo              dependencies       fmt                information        lock               overrides,         registry           specification      Upload
and                Cargo.toml         directory          for                init               lockfile           owner              Remove             specified          used
api                check              distributable      format             install            login              owners             reported           specified,         using
artifacts          Check              doc                from               Install            machine-readable   package            repository         stdin.             verify-project
as                 Checkout           documentation      fully              integration        main               package,           representation     tarball            version
Assemble           clean              documentation,     Generate           into               Manage             packages           resolved           test               versions
at                 clippy             errors             generated          is                 manifest           past               run                tests              warnings
Automatically      Compile            Execute            generate-lockfile  it                 manifest.          pkgid              Run                that               will
be                 concrete           existing           Git                its                metadata           Print              Rust               the                yank
bench              copy               fetch              git-checkout       JSON               network            publish            rustc              to
benchmarks         correctness        Fetch              has                lint               new                pushed             rustdoc            token
```

Output of `cargo version`:
cargo 1.31.0 (339d9f9c8 2018-11-16)

OS: Linux (bash) / Windows (Cygwin)

### PR DESCRIPTION
**Pull request by pavel-zhigulin**
Fixes #6461 -  cargo autocompletion script

### CODE REVIEW

**Comment by pavel-zhigulin** [comment]
Wrong issue number in commit(

File: src/etc/cargo.bashcomp.sh [hunk 1]
```diff
@@ -120,7 +120,7 @@ _cargo()
 } &&
 complete -F _cargo cargo
 
-__cargo_commands=$(cargo --list 2>/dev/null | tail -n +2)
+__cargo_commands=$(cargo --list 2>/dev/null | tail -n +2 | sed -E -e "s/^\s+//g" | cut -d" " -f1)
```
Comment (ehuss) [inline src/etc/cargo.bashcomp.sh]: This doesn't quite work on BSD-style systems like MacOS due to the \s.  I would use awk which I find to be a little more straightforward.

```suggestion
__cargo_commands=$(cargo --list 2>/dev/null | awk 'NR>1 {print $1}')
```

**Comment by pavel-zhigulin** [comment]
Updated

**Comment by dwijnand** [comment]
What was the original cause and how does this fix it?  I can't decipher it from the diff.

File: src/etc/cargo.bashcomp.sh [hunk 2]
```diff
@@ -120,7 +120,7 @@ _cargo()
 } &&
 complete -F _cargo cargo
 
-__cargo_commands=$(cargo --list 2>/dev/null | tail -n +2)
+__cargo_commands=$(cargo --list 2>/dev/null | tail -n +2 | awk 'NR>1 {print $1}')
```
Comment (ehuss) [inline src/etc/cargo.bashcomp.sh]: The `tail` is not necessary with the `NR` in awk.  Indeed, this is now skipping the first command (bench).

**Comment by ehuss** [comment]
> What was the original cause and how does this fix it? I can't decipher it from the diff.

This was caused by #5807.  It was getting confused by the descriptions.

**Comment by dwijnand** [comment]
Oh, that 5807 guy...

**Comment by pavel-zhigulin** [comment]
Remove `tail` call.

**Pull request merged**